test/integration/agent/openflow_test.go

```diff
@@ -422,6 +425,183 @@ func TestNetworkPolicyFlows(t *testing.T) {
+		k8stypes.NewEndpointInfo(&k8sproxy.BaseEndpointInfo{
+			Endpoint: net.JoinHostPort("10.20.0.11", "8081"),
+			IsLocal:  true,
+		}),
+		k8stypes.NewEndpointInfo(&k8sproxy.BaseEndpointInfo{
+			Endpoint: net.JoinHostPort("10.20.1.11", "8081"),
+			IsLocal:  false,
+		}),
+	}
+
+	stickyMaxAgeSeconds := uint16(30)
+
+	tcs := []struct {
+		svc       svcConfig
+		gid       uint32
+		endpoints []k8sproxy.Endpoint
+		stickyAge uint16
+	}{
+		{
+			svc: svcConfig{
+				protocol: ofconfig.ProtocolTCP,
+				ip:       net.ParseIP("10.20.30.41"),
+				port:     uint16(8000),
+			},
+			gid:       2,
+			endpoints: endpoints,
+			stickyAge: stickyMaxAgeSeconds,
+		},
+		{
+			svc: svcConfig{
+				protocol: ofconfig.ProtocolUDP,
+				ip:       net.ParseIP("10.20.30.42"),
+				port:     uint16(8000),
+			},
+			gid:       3,
+			endpoints: endpoints,
+			stickyAge: stickyMaxAgeSeconds,
+		},
+		{
+			svc: svcConfig{
+				protocol: ofconfig.ProtocolSCTP,
+				ip:       net.ParseIP("10.20.30.43"),
+				port:     uint16(8000),
+			},
+			gid:       4,
+			endpoints: endpoints,
+			stickyAge: stickyMaxAgeSeconds,
+		},
+	}
+
+	for _, tc := range tcs {
+		groupID := ofconfig.GroupIDType(tc.gid)
+		expTableFlows, expGroupBuckets := prepareProxyServiceGroupAndFlows(tc.gid, tc.svc, tc.endpoints, tc.stickyAge)
+		installServiceFlows(t, tc.gid, tc.svc, tc.endpoints, tc.stickyAge)
+		for _, tableFlow := range expTableFlows {
+			ofTestUtils.CheckFlowExists(t, ovsCtlClient, tableFlow.tableID, true, tableFlow.flows)
+		}
+		ofTestUtils.CheckGroupExists(t, ovsCtlClient, groupID, "select", expGroupBuckets, true)
+
+		uninstallServiceFlowsFunc(t, tc.gid, tc.svc, tc.endpoints)
+		for _, tableFlow := range expTableFlows {
+			ofTestUtils.CheckFlowExists(t, ovsCtlClient, tableFlow.tableID, false, tableFlow.flows)
+		}
+		ofTestUtils.CheckGroupExists(t, ovsCtlClient, groupID, "select", expGroupBuckets, false)
+	}
+
+}
+
+func installServiceFlows(t *testing.T, gid uint32, svc svcConfig, endpointList []k8sproxy.Endpoint, stickyMaxAgeSeconds uint16) {
+	groupID := ofconfig.GroupIDType(gid)
+	err := c.InstallEndpointFlows(svc.protocol, endpointList)
+	assert.Nil(t, err)
+	err = c.InstallServiceGroup(groupID, svc.withSessionAffinity, endpointList)
+	assert.Nil(t, err)
+	err = c.InstallServiceFlows(groupID, svc.ip, svc.port, svc.protocol, stickyMaxAgeSeconds)
+	assert.Nil(t, err)
+}
+
+func uninstallServiceFlowsFunc(t *testing.T, gid uint32, svc svcConfig, endpointList []k8sproxy.Endpoint) {
+	groupID := ofconfig.GroupIDType(gid)
+	err := c.UninstallServiceFlows(svc.ip, svc.port, svc.protocol)
+	assert.Nil(t, err)
+	err = c.UninstallServiceGroup(groupID)
+	assert.Nil(t, err)
+	for _, ep := range endpointList {
+		err := c.UninstallEndpointFlows(svc.protocol, ep)
+		assert.Nil(t, err)
+	}
+}
+
+func prepareProxyServiceGroupAndFlows(gid uint32, svc svcConfig, endpointList []k8sproxy.Endpoint, stickyAge uint16) (tableFlows []expectTableFlows, groupBuckets []string) {
```

Comment:
nit: "prepare" is probably not the best name here

maybe "expectedProxyServiceGroupAndFlows"?